In [1]:
import pandas as pd
from selenium.common.exceptions import NoSuchElementException
from selenium import webdriver
import time
import json

In [2]:
import sys
sys.path

['/Users/Thome/Desktop/CoderSchool/Shopee_Phone_EDA/EDA',
 '/Users/thome/.vscode/extensions/ms-toolsai.jupyter-2021.9.1001312534/pythonFiles',
 '/Users/thome/.vscode/extensions/ms-toolsai.jupyter-2021.9.1001312534/pythonFiles/lib/python',
 '/Users/thome/opt/anaconda3/envs/shopee_eda/lib/python39.zip',
 '/Users/thome/opt/anaconda3/envs/shopee_eda/lib/python3.9',
 '/Users/thome/opt/anaconda3/envs/shopee_eda/lib/python3.9/lib-dynload',
 '',
 '/Users/thome/opt/anaconda3/envs/shopee_eda/lib/python3.9/site-packages',
 '/Users/thome/opt/anaconda3/envs/shopee_eda/lib/python3.9/site-packages/IPython/extensions',
 '/Users/thome/.ipython']

In [3]:
# Để chạy jupyter trên vscode cần add python 3.8
sys.path.append('/Library/Python/3.8/site-packages')
sys.path

['/Users/Thome/Desktop/CoderSchool/Shopee_Phone_EDA/EDA',
 '/Users/thome/.vscode/extensions/ms-toolsai.jupyter-2021.9.1001312534/pythonFiles',
 '/Users/thome/.vscode/extensions/ms-toolsai.jupyter-2021.9.1001312534/pythonFiles/lib/python',
 '/Users/thome/opt/anaconda3/envs/shopee_eda/lib/python39.zip',
 '/Users/thome/opt/anaconda3/envs/shopee_eda/lib/python3.9',
 '/Users/thome/opt/anaconda3/envs/shopee_eda/lib/python3.9/lib-dynload',
 '',
 '/Users/thome/opt/anaconda3/envs/shopee_eda/lib/python3.9/site-packages',
 '/Users/thome/opt/anaconda3/envs/shopee_eda/lib/python3.9/site-packages/IPython/extensions',
 '/Users/thome/.ipython',
 '/Library/Python/3.8/site-packages']

# Clean data

## Overall smartphone industry on Shopee e-commerce

In [11]:
df = pd.read_csv('/Users/Thome/Desktop/CoderSchool/Shopee_Phone_EDA/EDA/shopee_phone.csv')
df.shape

(5760, 16)

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5760 entries, 0 to 5759
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Unnamed: 0        5760 non-null   int64  
 1   name              5585 non-null   object 
 2   price_dis         5585 non-null   object 
 3   price_nondis      5585 non-null   object 
 4   freeship          5760 non-null   bool   
 5   rating            5349 non-null   float64
 6   rating_count      5349 non-null   float64
 7   monthly_sold      2078 non-null   object 
 8   store_location    5585 non-null   object 
 9   monthly_event     5760 non-null   bool   
 10  favorite_product  1971 non-null   object 
 11  shopee_mall       5760 non-null   bool   
 12  discount_tag      1436 non-null   object 
 13  product_url       5585 non-null   object 
 14  image             5570 non-null   object 
 15  ShopDacBiet       5760 non-null   bool   
dtypes: bool(4), float64(2), int64(1), object(9

In [20]:
df.dropna(subset=['monthly_sold'], axis=0)

Unnamed: 0                                               name  \
0              0  Miếng dán điện thoại may mắn, lung linh phù hợ...   
1              1  Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...   
2              2                               Apple iPhone 11 64GB   
3              3  [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...   
4              4  Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...   
...          ...                                                ...   
4871        4871  ĐIỆN THOẠI SAMSUNG GALAXY NOTE 20 ULTRA [CHÍNH...   
5036        5036  LG V50 ThinQ 5G (6GB|128GB) ,Hàn Quốc,Snap 855...   
5133        5133                                     MIC NOKIA 6300   
5605        5605  Điện thoại Xiaomi Mi 11 Lite 5G Ram 8G/128G - ...   
5673        5673  Tai nghe iPhone, Tai phone chính hãng cho điện...   

       price_dis     price_nondis  freeship  rating  rating_count  \
0          1.400  ₫1.400 - ₫6.700      True    4.83         616.0   
1      3.390.000       ₫3.990.000      True    4.95         926.0   
2     13.990.000      ₫21.990.000      True    4.84        1132.0   
3        239.000         ₫239.000     False    5.00           4.0   
4      3.490.000       ₫3.490.000      True    4.93         715.0   
...          ...              ...       ...     ...           ...   
4871  29.990.000      ₫29.990.000     False     NaN           NaN   
5036   3.990.000       ₫3.990.000     False    4.33           3.0   
5133      19.000          ₫19.000     False    5.00           2.0   
5605   9.600.000       ₫9.600.000     False     NaN           NaN   
5673      35.000          ₫70.000     False    2.00           1.0   

     monthly_sold   store_location  monthly_event favorite_product  \
0      5,4k/tháng          Lào Cai          False        Yêu thích   
1      1,8k/tháng  TP. Hồ Chí Minh           True              NaN   
2      1,1k/tháng  TP. Hồ Chí Minh           True              NaN   
3       763/tháng         Hưng Yên          False              NaN   
4       715/tháng  TP. Hồ Chí Minh           True              NaN   
...           ...              ...            ...              ...   
4871      4/tháng  TP. Hồ Chí Minh           True        Yêu thích   
5036      1/tháng           Hà Nội           True        Yêu thích   
5133      1/tháng  TP. Hồ Chí Minh           True              NaN   
5605      1/tháng  TP. Hồ Chí Minh           True        Yêu thích   
5673      1/tháng        Thanh Hóa           True              NaN   

      shopee_mall discount_tag  \
0           False          NaN   
1            True    15%\nGIẢM   
2            True    36%\nGIẢM   
3           False          NaN   
4            True          NaN   
...           ...          ...   
4871        False          NaN   
5036        False          NaN   
5133        False          NaN   
5605        False          NaN   
5673        False    50%\nGIẢM   

                                            product_url  \
0     https://shopee.vn/Mi%E1%BA%BFng-d%C3%A1n-%C4%9...   
1     https://shopee.vn/%C4%90i%E1%BB%87n-tho%E1%BA%...   
2     https://shopee.vn/Apple-iPhone-11-64GB-i.88201...   
3     https://shopee.vn/-H%C3%A0ng-Ch%C3%ADnh-H%C3%A...   
4     https://shopee.vn/%C4%90i%E1%BB%87n-Tho%E1%BA%...   
...                                                 ...   
4871  https://shopee.vn/%C4%90I%E1%BB%86N-THO%E1%BA%...   
5036  https://shopee.vn/LG-V50-ThinQ-5G-(6GB-128GB)-...   
5133  https://shopee.vn/MIC-NOKIA-6300-i.406184606.8...   
5605  https://shopee.vn/%C4%90i%E1%BB%87n-tho%E1%BA%...   
5673  https://shopee.vn/Tai-nghe-iPhone-Tai-phone-ch...   

                                                  image  ShopDacBiet  
0     https://cf.shopee.vn/file/cd7f36d677941ad531a8...        False  
1     https://cf.shopee.vn/file/41694acce229f8102e75...        False  
2     https://cf.shopee.vn/file/b107f2e6167adb250100...        False  
3     https://cf.shopee.vn/file/5d35156b434e236ee552...        False  
4     http

Data được scrape từ 100 trang, mỗi trang gần 60 sản phẩm, tổng sản phầm scrape được trong tab bán chạy 5760
Trong đó monthly_sold nghĩa là số bán hàng tháng chỉ có 2078 thông tin. Vì tập trung focus nên chúng ta drop hết các nan value theo cột số bán hàng tháng này

In [21]:
df_overall = pd.read_csv('/Users/Thome/Desktop/CoderSchool/Shopee_Phone_EDA/EDA/shopee_phone_final.csv')

In [22]:
df_overall.shape

(2078, 17)

In [23]:
df_overall.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2078 entries, 0 to 2077
Data columns (total 17 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Unnamed: 0        2078 non-null   int64  
 1   Unnamed: 0.1      2078 non-null   int64  
 2   name              2078 non-null   object 
 3   price_dis         2078 non-null   object 
 4   price_nondis      2078 non-null   object 
 5   freeship          2078 non-null   bool   
 6   rating            1791 non-null   float64
 7   rating_count      1791 non-null   float64
 8   monthly_sold      2078 non-null   object 
 9   store_location    2078 non-null   object 
 10  monthly_event     2078 non-null   bool   
 11  favorite_product  770 non-null    object 
 12  shopee_mall       2078 non-null   bool   
 13  discount_tag      755 non-null    object 
 14  product_url       2078 non-null   object 
 15  image             2063 non-null   object 
 16  ShopDacBiet       2078 non-null   bool   


In [28]:
df_overall = df_overall.drop(columns=['Unnamed: 0',	'Unnamed: 0.1'])

In [29]:
df_overall.head()

name   price_dis  \
0  Miếng dán điện thoại may mắn, lung linh phù hợ...       1.400   
1  Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...   3.390.000   
2                               Apple iPhone 11 64GB  13.990.000   
3  [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...     239.000   
4  Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...   3.490.000   

      price_nondis  freeship  rating  rating_count monthly_sold  \
0  ₫1.400 - ₫6.700      True    4.83         616.0   5,4k/tháng   
1       ₫3.990.000      True    4.95         926.0   1,8k/tháng   
2      ₫21.990.000      True    4.84        1132.0   1,1k/tháng   
3         ₫239.000     False    5.00           4.0    763/tháng   
4       ₫3.490.000      True    4.93         715.0    715/tháng   

    store_location  monthly_event favorite_product  shopee_mall discount_tag  \
0          Lào Cai          False        Yêu thích        False          NaN   
1  TP. Hồ Chí Minh           True              NaN         True    15%\nGIẢM   
2  TP. Hồ Chí Minh           True              NaN         True    36%\nGIẢM   
3         Hưng Yên          False              NaN        False          NaN   
4  TP. Hồ Chí Minh           True              NaN         True          NaN   

                                         product_url  \
0  https://shopee.vn/Mi%E1%BA%BFng-d%C3%A1n-%C4%9...   
1  https://shopee.vn/%C4%90i%E1%BB%87n-tho%E1%BA%...   
2  https://shopee.vn/Apple-iPhone-11-64GB-i.88201...   
3  https://shopee.vn/-H%C3%A0ng-Ch%C3%ADnh-H%C3%A...   
4  https://shopee.vn/%C4%90i%E1%BB%87n-Tho%E1%BA%...   

                                               image  ShopDacBiet  
0  https://cf.shopee.vn/file/cd7f36d677941ad531a8...        False  
1  https://cf.shopee.vn/file/41694acce229f8102e75...        False  
2  https://cf.shopee.vn/file/b107f2e6167adb250100...        False  
3  https://cf.shopee.vn/file/5d35156b434e236ee552...        False  
4  https://cf.shopee.vn/file/a277200d45df2ef46c4a...        False

In [32]:
df_overall.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2078 entries, 0 to 2077
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   name              2078 non-null   object 
 1   price_dis         2078 non-null   object 
 2   price_nondis      2078 non-null   object 
 3   freeship          2078 non-null   bool   
 4   rating            1791 non-null   float64
 5   rating_count      1791 non-null   float64
 6   monthly_sold      2078 non-null   object 
 7   store_location    2078 non-null   object 
 8   monthly_event     2078 non-null   bool   
 9   favorite_product  770 non-null    object 
 10  shopee_mall       2078 non-null   bool   
 11  discount_tag      755 non-null    object 
 12  product_url       2078 non-null   object 
 13  image             2063 non-null   object 
 14  ShopDacBiet       2078 non-null   bool   
dtypes: bool(4), float64(2), object(9)
memory usage: 186.8+ KB


In [33]:
df_overall['name'].value_counts

<bound method IndexOpsMixin.value_counts of 0       Miếng dán điện thoại may mắn, lung linh phù hợ...
1       Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...
2                                    Apple iPhone 11 64GB
3       [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...
4       Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...
                              ...                        
2073    ĐIỆN THOẠI SAMSUNG GALAXY NOTE 20 ULTRA [CHÍNH...
2074    LG V50 ThinQ 5G (6GB|128GB) ,Hàn Quốc,Snap 855...
2075                                       MIC NOKIA 6300
2076    Điện thoại Xiaomi Mi 11 Lite 5G Ram 8G/128G - ...
2077    Tai nghe iPhone, Tai phone chính hãng cho điện...
Name: name, Length: 2078, dtype: object>

In [31]:
df_overall['favorite_product'].value_counts()

Yêu thích    770
Name: favorite_product, dtype: int64

In [35]:
df_overall.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2078 entries, 0 to 2077
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   name              2078 non-null   object 
 1   price_dis         2078 non-null   object 
 2   price_nondis      2078 non-null   object 
 3   freeship          2078 non-null   bool   
 4   rating            1791 non-null   float64
 5   rating_count      1791 non-null   float64
 6   monthly_sold      2078 non-null   object 
 7   store_location    2078 non-null   object 
 8   monthly_event     2078 non-null   bool   
 9   favorite_product  770 non-null    object 
 10  shopee_mall       2078 non-null   bool   
 11  discount_tag      755 non-null    object 
 12  product_url       2078 non-null   object 
 13  image             2063 non-null   object 
 14  ShopDacBiet       2078 non-null   bool   
dtypes: bool(4), float64(2), object(9)
memory usage: 186.8+ KB


Merge detailed data

In [40]:
detailed_1 = pd.read_csv('/Users/Thome/Desktop/CoderSchool/Shopee_Phone_EDA/EDA/detailed_data_1.csv')
detailed_2 = pd.read_csv('/Users/Thome/Desktop/CoderSchool/Shopee_Phone_EDA/EDA/detailed_data_2.csv')
detailed_3 = pd.read_csv('/Users/Thome/Desktop/CoderSchool/Shopee_Phone_EDA/EDA/detailed_data_3.csv')
detailed_4 = pd.read_csv('/Users/Thome/Desktop/CoderSchool/Shopee_Phone_EDA/EDA/detailed_data_4.csv')
detailed_5 = pd.read_csv('/Users/Thome/Desktop/CoderSchool/Shopee_Phone_EDA/EDA/detailed_data_5.csv')
detailed_6 = pd.read_csv('/Users/Thome/Desktop/CoderSchool/Shopee_Phone_EDA/EDA/detailed_data_6.csv')
detailed_7 = pd.read_csv('/Users/Thome/Desktop/CoderSchool/Shopee_Phone_EDA/EDA/detailed_data_7.csv')


In [144]:
frames = [detailed_1, detailed_2, detailed_3, detailed_4, detailed_5, detailed_6, detailed_7]
detailed_data = pd.concat(frames)

In [191]:
name_detailed = detailed_data['name_product'].reset_index()
name_overall = df_overall['name'].reset_index()
concate_name = [name_detailed, name_overall]


In [410]:
detailed_data.T

0    \
Unnamed: 0                                                            0   
name_product          Yêu Thích\nMiếng dán điện thoại may mắn, lung ...   
rating_product                                                     4.83   
rating_count_product                                              632.0   
quantity_5_star                                                      81   
quantity_4_star                                                    11.0   
quantity_3_star                                                     NaN   
quantity_2_star                                                     NaN   
quantity_1_star                                                     NaN   
count_cmt                                                           360   
images_video                                                      293.0   
sold_out                                                          15,5k   
low_price                                                        6700.0   
hight_price                                                      1400.0   
price                                                               NaN   
inventory                                                         22920   
return_7_days                                                       NaN   
authentic                                                           NaN   
freeship                                                            NaN   
shopee_guaranteed                           3 Ngày Trả Hàng / Hoàn Tiền   
company                                             Tổng kho hàng China   
shop_name                                               hung_pham130217   
shopee_mall                                                       False   
shop_total_review                                                  1,5k   
total_products                                                       13   
response_rate                                                       96%   
time_response                                            trong vài phút   
years_ex                                                 27 tháng trước   
follower                                                           3,2k   

                                                                    1    \
Unnamed: 0                                                            1   
name_product          [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...   
rating_product                                                     4.95   
rating_count_product                                              940.0   
quantity_5_star                                                     916   
quantity_4_star                                                     7.0   
quantity_3_star                                                     NaN   
quantity_2_star                                                     NaN   
quantity_1_star                                                     NaN   
count_cmt                                                           394   
images_video                                                      292.0   
sold_out                                                           4,7k   
low_price                                                           NaN   
hight_price                                                         NaN   
price                                                         3790000.0   
inventory                                                            59   
return_7_days                                  7 ngày miễn phí trả hàng   
authentic                                          Hàng chính hãng 100%   
freeship                                            Miễn phí vận chuyển   
shopee_guaranteed                                                   NaN   
company                                             VIVO OFFICIAL STORE   
shop_name                                        vivo_official_store_vn   
shopee_mall                                                        True   
shop_total_review

In [346]:
df_overall.T

0     \
name              Miếng dán điện thoại may mắn, lung linh phù hợ...   
price_dis                                                     1.400   
price_nondis                                        ₫1.400 - ₫6.700   
freeship                                                       True   
rating                                                         4.83   
rating_count                                                  616.0   
monthly_sold                                             5,4k/tháng   
store_location                                              Lào Cai   
monthly_event                                                 False   
favorite_product                                          Yêu thích   
shopee_mall                                                   False   
discount_tag                                                    NaN   
product_url       https://shopee.vn/Mi%E1%BA%BFng-d%C3%A1n-%C4%9...   
image             https://cf.shopee.vn/file/cd7f36d677941ad531a8...   
ShopDacBiet                                                   False   

                                                               1     \
name              Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...   
price_dis                                                 3.390.000   
price_nondis                                             ₫3.990.000   
freeship                                                       True   
rating                                                         4.95   
rating_count                                                  926.0   
monthly_sold                                             1,8k/tháng   
store_location                                      TP. Hồ Chí Minh   
monthly_event                                                  True   
favorite_product                                                NaN   
shopee_mall                                                    True   
discount_tag                                              15%\nGIẢM   
product_url       https://shopee.vn/%C4%90i%E1%BB%87n-tho%E1%BA%...   
image             https://cf.shopee.vn/file/41694acce229f8102e75...   
ShopDacBiet                                                   False   

                                                               2     \
name                                           Apple iPhone 11 64GB   
price_dis                                                13.990.000   
price_nondis                                            ₫21.990.000   
freeship                                                       True   
rating                                                         4.84   
rating_count                                                 1132.0   
monthly_sold                                             1,1k/tháng   
store_location                                      TP. Hồ Chí Minh   
monthly_event                                                  True   
favorite_product                                                NaN   
shopee_mall                                                    True   
discount_tag                                              36%\nGIẢM   
product_url       https://shopee.vn/Apple-iPhone-11-64GB-i.88201...   
image             https://cf.shopee.vn/file/b107f2e6167adb250100...   
ShopDacBiet                                                   False   

                                                               3     \
name              [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...   
price_dis                                                   239.000   
price_nondis                                               ₫239.000   
freeship                                                      False   
rating                                                          5.0   
rating_count                                                    4.0   
monthly_sold                                              763/tháng   
store_location                                             Hưng Yên   
monthly_event

In [192]:
name_detailed

index                                       name_product
0         0  Yêu Thích\nMiếng dán điện thoại may mắn, lung ...
1         1  [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...
2         2                               Apple iPhone 11 64GB
3         3  [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...
4         4  Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...
...     ...                                                ...
1797    107  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] ĐI...
1798    108  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] LG...
1799    109                                     MIC NOKIA 6300
1800    110  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] Đi...
1801    111  Tai nghe iPhone, Tai phone chính hãng cho điện...

[1802 rows x 2 columns]

In [225]:
compare_name = pd.concat(concate_name, axis=1, ignore_index=True)

In [175]:
name_detailed.shape

(1802, 2)

In [176]:
name_overall.shape

(2078, 2)

In [226]:
compare_name.drop(columns=[0, 2], inplace=True)

In [227]:
compare_name.rename(columns={1:'name_detail', 3:'name_overall'}, inplace=True)

In [228]:
compare_name.dropna(inplace=True)

In [229]:
compare_name['compare'] = compare_name['name_overall'] == compare_name['name_detail']

In [249]:
compare_name['compare'].value_counts()

False    1366
True      436
Name: compare, dtype: int64

In [230]:
compare_name[compare_name['compare'] == False]

name_detail  \
0     Yêu Thích\nMiếng dán điện thoại may mắn, lung ...   
1     [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...   
5     Yêu Thích\nĐiện thoại Mini - L8STAR BM10 - ( N...   
6     Yêu Thích\nĐiện Thoại Nokia 101,Nokia 100, Nok...   
7     [Mã 1010ELSALE1 giảm 5% đơn 3TR] Điện thoại OP...   
...                                                 ...   
1797  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] ĐI...   
1798  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] LG...   
1799                                     MIC NOKIA 6300   
1800  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] Đi...   
1801  Tai nghe iPhone, Tai phone chính hãng cho điện...   

                                           name_overall  compare  
0     Miếng dán điện thoại may mắn, lung linh phù hợ...    False  
1     Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...    False  
5     Điện thoại Mini - L8STAR BM10 - ( Nokia 3310 )...    False  
6     Điện Thoại Nokia 101,Nokia 100, Nokia 105 Zin ...    False  
7             Điện thoại OPPO A16 4GB - Hàng chính hãng    False  
...                                                 ...      ...  
1797  Điện Thoại Samsung Galaxy S20 FE 5G nguyên Zin...    False  
1798   Điện Thoại Mini 4G Palm Phone PVG100 New Sealbox    False  
1799                     xác điện thoại oppo a3s vỡ màn    False  
1800               Nắp Lưng Điện Thoại Xiaomi Mi 8 Lite    False  
1801  Điện thoại LG G7 ThinQ,Ram 4/64gb,Snap 845,LCD...    False  

[1366 rows x 3 columns]

làm name_detail sao cho == name_overall

In [260]:
compare_name_copy = compare_name.copy()

In [263]:
compare_name_copy

name_detail  \
0     Yêu Thích\nMiếng dán điện thoại may mắn, lung ...   
1     [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...   
2                                  Apple iPhone 11 64GB   
3     [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...   
4     Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...   
...                                                 ...   
1797  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] ĐI...   
1798  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] LG...   
1799                                     MIC NOKIA 6300   
1800  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] Đi...   
1801  Tai nghe iPhone, Tai phone chính hãng cho điện...   

                                           name_overall  compare  \
0     Miếng dán điện thoại may mắn, lung linh phù hợ...    False   
1     Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...    False   
2                                  Apple iPhone 11 64GB     True   
3     [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...     True   
4     Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...     True   
...                                                 ...      ...   
1797  Điện Thoại Samsung Galaxy S20 FE 5G nguyên Zin...    False   
1798   Điện Thoại Mini 4G Palm Phone PVG100 New Sealbox    False   
1799                     xác điện thoại oppo a3s vỡ màn    False   
1800               Nắp Lưng Điện Thoại Xiaomi Mi 8 Lite    False   
1801  Điện thoại LG G7 ThinQ,Ram 4/64gb,Snap 845,LCD...    False   

     name_detail_2  compare_af_fix  
0             None           False  
1             None           False  
2             None           False  
3             None           False  
4             None           False  
...            ...             ...  
1797          None           False  
1798          None           False  
1799          None           False  
1800          None           False  
1801          None           False  

[1802 rows x 5 columns]

In [264]:
def fix_name(series_name):
    #for i in series_name:
    if '\n' in series_name:
        series_name = series_name.split('\n')[-1]
    else:
        pass
    if '3TR]' in series_name:
        series_name = series_name.split('3TR]')[-1]
    else:
        pass
    return series_name

            
compare_name_copy['name_detail_2'] = compare_name_copy['name_detail'].apply(fix_name)

In [265]:
compare_name_copy['compare_af_fix'] = compare_name_copy['name_detail_2'] == compare_name_copy['name_overall']
compare_name_copy['compare_af_fix'].value_counts()

False    1058
True      744
Name: compare_af_fix, dtype: int64

Lấn 1:  False    1366
        True      436

Lần 2:  False    1058
        True      744

In [ ]:
a.split('TR]')

['[Mã 1010ELSALE1 giảm 5% đơn 3',
 ' ĐIỆN THOẠI SAMSUNG GALAXY NOTE 20 ULTRA [CHÍNH HÃNG NGUYÊN SEAL]']

In [254]:
compare_name_copy[compare_name_copy['compare_af_fix'] == False]

name_detail  \
1     [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...   
7     [Mã 1010ELSALE1 giảm 5% đơn 3TR] Điện thoại OP...   
13    Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] Đi...   
14    [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...   
24    Yêu Thích\nĐiện Thoại Realme Q3 Pro 5G chip Di...   
...                                                 ...   
1797  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] ĐI...   
1798  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] LG...   
1799                                     MIC NOKIA 6300   
1800  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] Đi...   
1801  Tai nghe iPhone, Tai phone chính hãng cho điện...   

                                           name_overall  compare  \
1     Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...    False   
7             Điện thoại OPPO A16 4GB - Hàng chính hãng    False   
13    Điện Thoại Xiaomi Redmi K40 Gaming Edition - h...    False   
14     Điện thoại Vivo Y1s 2GB + 32GB - Hàng chính hãng    False   
24    Điện Thoại Realme Q3 Pro 6/128, 8/128GB Khang ...    False   
...                                                 ...      ...   
1797  Điện Thoại Samsung Galaxy S20 FE 5G nguyên Zin...    False   
1798   Điện Thoại Mini 4G Palm Phone PVG100 New Sealbox    False   
1799                     xác điện thoại oppo a3s vỡ màn    False   
1800               Nắp Lưng Điện Thoại Xiaomi Mi 8 Lite    False   
1801  Điện thoại LG G7 ThinQ,Ram 4/64gb,Snap 845,LCD...    False   

                                          name_detail_2  compare_af_fix  
1     [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...           False  
7             Điện thoại OPPO A16 4GB - Hàng chính hãng           False  
13     Điện Thoại Xiaomi Redmi K40 Gaming Edition - ...           False  
14    [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...           False  
24    Điện Thoại Realme Q3 Pro 5G chip Dimensity 110...           False  
...                                                 ...             ...  
1797   ĐIỆN THOẠI SAMSUNG GALAXY NOTE 20 ULTRA [CHÍN...           False  
1798   LG V50 ThinQ 5G (6GB|128GB) ,Hàn Quốc,Snap 85...           False  
1799                                     MIC NOKIA 6300           False  
1800   Điện thoại Xiaomi Mi 11 Lite 5G Ram 8G/128G -...           False  
1801  Tai nghe iPhone, Tai phone chính hãng cho điện...           False  

[1058 rows x 5 columns]

In [313]:
def fix_name_2(row):
    if row['compare_af_fix'] == False:
        if "K]" in row['name_detail_2']:
            row['name_detail_2'] = row['name_detail_2'].split("K]")[-1]
        else:
            pass

    
    return row['name_detail_2']

compare_name_copy['name_detail_3'] = compare_name_copy.apply(fix_name_2, axis=1)
   


In [315]:
compare_name_copy

name_detail  \
0     Yêu Thích\nMiếng dán điện thoại may mắn, lung ...   
1     [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...   
2                                  Apple iPhone 11 64GB   
3     [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...   
4     Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...   
...                                                 ...   
1797  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] ĐI...   
1798  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] LG...   
1799                                     MIC NOKIA 6300   
1800  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] Đi...   
1801  Tai nghe iPhone, Tai phone chính hãng cho điện...   

                                           name_overall  compare  \
0     Miếng dán điện thoại may mắn, lung linh phù hợ...    False   
1     Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...    False   
2                                  Apple iPhone 11 64GB     True   
3     [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...     True   
4     Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...     True   
...                                                 ...      ...   
1797  Điện Thoại Samsung Galaxy S20 FE 5G nguyên Zin...    False   
1798   Điện Thoại Mini 4G Palm Phone PVG100 New Sealbox    False   
1799                     xác điện thoại oppo a3s vỡ màn    False   
1800               Nắp Lưng Điện Thoại Xiaomi Mi 8 Lite    False   
1801  Điện thoại LG G7 ThinQ,Ram 4/64gb,Snap 845,LCD...    False   

                                          name_detail_2  compare_af_fix  \
0     Miếng dán điện thoại may mắn, lung linh phù hợ...            True   
1     [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...           False   
2                                  Apple iPhone 11 64GB            True   
3     [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...            True   
4     Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...            True   
...                                                 ...             ...   
1797   ĐIỆN THOẠI SAMSUNG GALAXY NOTE 20 ULTRA [CHÍN...           False   
1798   LG V50 ThinQ 5G (6GB|128GB) ,Hàn Quốc,Snap 85...           False   
1799                                     MIC NOKIA 6300           False   
1800   Điện thoại Xiaomi Mi 11 Lite 5G Ram 8G/128G -...           False   
1801  Tai nghe iPhone, Tai phone chính hãng cho điện...           False   

                                          name_detail_3  compare_af_fix_2  
0     Miếng dán điện thoại may mắn, lung linh phù hợ...              True  
1      Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng ...             False  
2                                  Apple iPhone 11 64GB              True  
3     [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...              True  
4     Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...              True  
...                                                 ...               ...  
1797   ĐIỆN THOẠI SAMSUNG GALAXY NOTE 20 ULTRA [CHÍN...             False  
1798   LG V50 ThinQ 5G (6GB|128GB) ,Hàn Quốc,Snap 85...             False  
1799                                     MIC NOKIA 6300             False  
1800   Điện thoại Xiaomi Mi 11 Lite 5G Ram 8G/128G -...             False  
1801  Tai nghe iPhone, Tai phone chính hãng cho điện...             False  

[1802 rows x 7 columns]

false Khoảng trắng đầu trong tên sản phẩm

In [320]:
def fix_name_3(series):
    if series[0:1] == ' ':
        series = series[1:]
    else:
        series
    return series

compare_name_copy['name_detail_3'] = compare_name_copy['name_detail_3'].apply(fix_name_3)

In [321]:
compare_name_copy['compare_af_fix_2'] = compare_name_copy['name_detail_3'] == compare_name_copy['name_overall']
compare_name_copy['compare_af_fix_2'].value_counts()

True     939
False    863
Name: compare_af_fix_2, dtype: int64

Lấn 1:  False    1366
        True      436

Lần 2:  False    1058
        True      744

lần 3:  True     939
        False    863    

In [ ]:
a.split('TR]')

['[Mã 1010ELSALE1 giảm 5% đơn 3',
 ' ĐIỆN THOẠI SAMSUNG GALAXY NOTE 20 ULTRA [CHÍNH HÃNG NGUYÊN SEAL]']

In [324]:
compare_name_copy.head()

name_detail  \
0  Yêu Thích\nMiếng dán điện thoại may mắn, lung ...   
1  [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...   
2                               Apple iPhone 11 64GB   
3  [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...   
4  Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...   

                                        name_overall  compare  \
0  Miếng dán điện thoại may mắn, lung linh phù hợ...    False   
1  Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...    False   
2                               Apple iPhone 11 64GB     True   
3  [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...     True   
4  Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...     True   

                                       name_detail_2  compare_af_fix  \
0  Miếng dán điện thoại may mắn, lung linh phù hợ...            True   
1  [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...           False   
2                               Apple iPhone 11 64GB            True   
3  [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...            True   
4  Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...            True   

                                       name_detail_3  compare_af_fix_2  
0  Miếng dán điện thoại may mắn, lung linh phù hợ...              True  
1  Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...              True  
2                               Apple iPhone 11 64GB              True  
3  [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...              True  
4  Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...              True

In [330]:
compare_name_copy.columns

Index(['name_detail', 'name_overall', 'compare', 'name_detail_2',
       'compare_af_fix', 'name_detail_3', 'compare_af_fix_2'],
      dtype='object')

In [337]:
compare_name_copy.head()

name_overall  \
0  miếng dán điện thoại may mắn, lung linh phù hợ...   
1  điện thoại vivo y20 (2021) 4gb + 64gb - hàng c...   
2                               apple iphone 11 64gb   
3  [hàng chính hãng 100%] nokia 1280, điện thoại ...   
4  điện thoại samsung galaxy m12 (3gb/32gb) - hãn...   

                                       name_detail_3  compare_af_fix_2  
0  miếng dán điện thoại may mắn, lung linh phù hợ...              True  
1  điện thoại vivo y20 (2021) 4gb + 64gb - hàng c...              True  
2                               apple iphone 11 64gb              True  
3  [hàng chính hãng 100%] nokia 1280, điện thoại ...              True  
4  điện thoại samsung galaxy m12 (3gb/32gb) - hãn...              True

In [333]:
compare_name_copy.drop(columns=['name_detail', 'compare', 'name_detail_2', 'compare_af_fix'], inplace=True)

In [336]:
def lower_name(series):
    return series.lower()

compare_name_copy['name_overall'] = compare_name_copy['name_overall'].apply(lower_name)
compare_name_copy['name_detail_3'] = compare_name_copy['name_detail_3'].apply(lower_name)

apply regex after pattern: điện thoại .+

In [338]:
import re

In [ ]:
def regex(series):
    pattern = r'điện thoại .+'
    check = re.findall(pattern, series)

    if len(check) == 0:
        return None  
    else:
        return check[0]
          

compare_name_copy['name_detail_3'] = compare_name_copy['name_detail_3'].apply(regex)
compare_name_copy['name_overall'] = compare_name_copy['name_overall'].apply(regex)

In [406]:
compare_name_copy['compare_af_fix_3'] = compare_name_copy['name_detail_3'] == compare_name_copy['name_overall']
compare_name_copy['compare_af_fix_3'].value_counts()

False    984
True     818
Name: compare_af_fix_3, dtype: int64

Lấn 1:  False    1366
        True      436

Lần 2:  False    1058
        True      744

lần 3:  True     939
        False    863

Lần 4:  True     984
        False    818

In [398]:
compare_name_copy

name_overall  \
0     điện thoại may mắn, lung linh phù hợp các dòng...   
1     điện thoại vivo y20 (2021) 4gb + 64gb - hàng c...   
2                                                  None   
3     điện thoại nokia1280 zin chính hãng đủ phụ kiệ...   
4     điện thoại samsung galaxy m12 (3gb/32gb) - hãn...   
...                                                 ...   
1797  điện thoại samsung galaxy s20 fe 5g nguyên zin...   
1798   điện thoại mini 4g palm phone pvg100 new sealbox   
1799                         điện thoại oppo a3s vỡ màn   
1800                        điện thoại xiaomi mi 8 lite   
1801  điện thoại lg g7 thinq,ram 4/64gb,snap 845,lcd...   

                                          name_detail_3  compare_af_fix_2  \
0     điện thoại may mắn, lung linh phù hợp các dòng...              True   
1     điện thoại vivo y20 (2021) 4gb + 64gb - hàng c...              True   
2                                                  None              True   
3     điện thoại nokia1280 zin chính hãng đủ phụ kiệ...              True   
4     điện thoại samsung galaxy m12 (3gb/32gb) - hãn...              True   
...                                                 ...               ...   
1797  điện thoại samsung galaxy note 20 ultra [chính...             False   
1798                                               None             False   
1799                                               None             False   
1800  điện thoại xiaomi mi 11 lite 5g ram 8g/128g - ...             False   
1801                               điện thoại - beetech             False   

      compare_af_fix_3  
0                 True  
1                 True  
2                False  
3                 True  
4                 True  
...                ...  
1797             False  
1798             False  
1799             False  
1800             False  
1801             False  

[1802 rows x 4 columns]

In [399]:
compare_name_copy[compare_name_copy['compare_af_fix_3'] == False]

name_overall  \
2                                                  None   
17                                                 None   
18                                                 None   
19                                                 None   
24    điện thoại realme q3 pro 6/128, 8/128gb khang ...   
...                                                 ...   
1797  điện thoại samsung galaxy s20 fe 5g nguyên zin...   
1798   điện thoại mini 4g palm phone pvg100 new sealbox   
1799                         điện thoại oppo a3s vỡ màn   
1800                        điện thoại xiaomi mi 8 lite   
1801  điện thoại lg g7 thinq,ram 4/64gb,snap 845,lcd...   

                                          name_detail_3  compare_af_fix_2  \
2                                                  None              True   
17                                                 None              True   
18                                                 None              True   
19                                                 None              True   
24    điện thoại realme q3 pro 5g chip dimensity 110...             False   
...                                                 ...               ...   
1797  điện thoại samsung galaxy note 20 ultra [chính...             False   
1798                                               None             False   
1799                                               None             False   
1800  điện thoại xiaomi mi 11 lite 5g ram 8g/128g - ...             False   
1801                               điện thoại - beetech             False   

      compare_af_fix_3  
2                False  
17               False  
18               False  
19               False  
24               False  
...                ...  
1797             False  
1798             False  
1799             False  
1800             False  
1801             False  

[984 rows x 4 columns]

In [407]:
for i in compare_name_copy[compare_name_copy['compare_af_fix_3'] == False]['name_overall']:
    if i != None:
        print(i)

điện thoại realme q3 pro 6/128, 8/128gb khang nhung - hàng nhập khẩu
điện thoại vivo y55 game mượt - snap 430 màn 5.2
điện thoại realme gt neo flash dimensity 1200 5g khang nhung - hàng nhập khẩu
điện thoại vivo iqoo z1x 6/64gb khang nhung - hàng nhập khẩu
điện thoại xiaomi mi 11 lite 5g chip snapdragon 780g khang nhung - hàng nhập khẩu
điện thoại xiaomi black shark 4 6/128, 8/128gb khang nhung - hàng nhập khẩu
điện thoại di động nokia 110 4g ta - 1376 ds vn
điện thoại mini i8, siêu nhỏ thiết kế giống iphone x
điện thoại nokia 1110i zin bh 6 tháng
điện thoại xiaomi redmi note 10s (8gb/128gb) - hàng chính hãng
điện thoại xiaomi redmi 9 (3gb/32gb) - hàng chính hãng
điện thoại vsmart aris (8gb/128gb) - hàng chính hãng
điện thoại tài lộc, buôn may bán đắt phù hợp các dòng điện thoại, đẹp lung linh.
điện thoại realme c2( vỏ + xương )
điện thoại nokia 106 2 sim (2020) đủ pin sạc bảo hành 6 tháng
điện thoại samsung s21 s21 plus s21 ultra hiệu dux ducis hàng chuẩn full box
điện thoại lg v50s r

In [408]:
for j in compare_name_copy[compare_name_copy['compare_af_fix_3'] == False]['name_detail_3']:
    if j != None:
        print(j)

điện thoại realme q3 pro 5g chip dimensity 1100 pin 4500 - sạc nhanh 30w khang nhung - hàng nhập khẩu
điện thoại vivo y55 game mượt - snap 430 màn 5.2 inch
điện thoại realme gt neo flash dimensity 1200 pin 4500 sạc nhanh 65w khang nhung - hàng nhập khẩu
điện thoại vivo iqoo z1x snapdragon 765 pin pin 500 sạc nhanh 33w khang nhung - hàng nhập khẩu
điện thoại xiaomi mi 11 lite 5g chip snapdragon 780g pin 4250 khang nhung - hàng nhập khẩu
điện thoại xiaomi black shark - snapdragon 870 pin 4500 sạc nhanh 120w khang nhung - hàng nhập khẩu
điện thoại di động nokia 110 4g ta-1376 ds vn xanh (aqua)_16lyre21a02aq_2021
điện thoại mini i8, siêu nhỏ thiết kế nhỏ gọn
điện thoại nokia 1110i zin bh 6 tháng
điện thoại xiaomi redmi note 10s (8gb/128gb) - hàng chính hãng
điện thoại xiaomi redmi 9 (3gb/32gb) - hàng chính hãng
điện thoại vsmart aris (8gb/128gb) - hàng chính hãng
điện thoại tài lộc, buôn may bán đắt phù hợp các dòng điện thoại, đẹp lung linh.
điện thoại realme c2( vỏ + xương )
điện thoại n

Chốt phương án 1000 sản phẩm, tương đương 50 trang đầu tiên

In [491]:
# def function clean name của 2 bảng

def clean_name(series):
    series_name =   series.lower()

    if '\n' in series_name:
        series_name = series_name.split('\n')[-1]

    if "tr]" in series_name:
        series_name = series_name.split("tr]")[-1]

    if "k]" in series_name:
        series_name = series_name.split("k]")[-1]

    if series_name[0:1] == ' ':
        series_name = series_name[1:]

    pattern = r'điện thoại .+'
    check = re.findall(pattern, series_name)
    if len(check) == 0:
        return series_name
    else:
        return check[0]

In [492]:
compare_name_copy_2 = compare_name.copy()

In [473]:
compare_name_copy_3 = compare_name.copy()

Test apply

In [474]:
compare_name_copy_3

name_detail  \
0     Yêu Thích\nMiếng dán điện thoại may mắn, lung ...   
1     [Mã ELMALL300K giảm 5% đơn 500K] Điện thoại Vi...   
2                                  Apple iPhone 11 64GB   
3     [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...   
4     Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...   
...                                                 ...   
1797  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] ĐI...   
1798  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] LG...   
1799                                     MIC NOKIA 6300   
1800  Yêu Thích\n[Mã 1010ELSALE1 giảm 5% đơn 3TR] Đi...   
1801  Tai nghe iPhone, Tai phone chính hãng cho điện...   

                                           name_overall  compare  
0     Miếng dán điện thoại may mắn, lung linh phù hợ...    False  
1     Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...    False  
2                                  Apple iPhone 11 64GB     True  
3     [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...     True  
4     Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...     True  
...                                                 ...      ...  
1797  Điện Thoại Samsung Galaxy S20 FE 5G nguyên Zin...    False  
1798   Điện Thoại Mini 4G Palm Phone PVG100 New Sealbox    False  
1799                     xác điện thoại oppo a3s vỡ màn    False  
1800               Nắp Lưng Điện Thoại Xiaomi Mi 8 Lite    False  
1801  Điện thoại LG G7 ThinQ,Ram 4/64gb,Snap 845,LCD...    False  

[1802 rows x 3 columns]

In [480]:
compare_name_copy_2

name_detail  \
0     điện thoại may mắn, lung linh phù hợp các dòng...   
1     điện thoại vivo y20 (2021) 4gb + 64gb - hàng c...   
2                                                  None   
3     điện thoại nokia1280 zin chính hãng đủ phụ kiệ...   
4     điện thoại samsung galaxy m12 (3gb/32gb) - hãn...   
...                                                 ...   
1797  điện thoại samsung galaxy note 20 ultra [chính...   
1798                                               None   
1799                                               None   
1800  điện thoại xiaomi mi 11 lite 5g ram 8g/128g - ...   
1801                               điện thoại - beetech   

                                           name_overall  compare  compare_2  
0     điện thoại may mắn, lung linh phù hợp các dòng...    False       True  
1     điện thoại vivo y20 (2021) 4gb + 64gb - hàng c...    False       True  
2                                                  None     True      False  
3     điện thoại nokia1280 zin chính hãng đủ phụ kiệ...     True       True  
4     điện thoại samsung galaxy m12 (3gb/32gb) - hãn...     True       True  
...                                                 ...      ...        ...  
1797  điện thoại samsung galaxy s20 fe 5g nguyên zin...    False      False  
1798   điện thoại mini 4g palm phone pvg100 new sealbox    False      False  
1799                         điện thoại oppo a3s vỡ màn    False      False  
1800                        điện thoại xiaomi mi 8 lite    False      False  
1801  điện thoại lg g7 thinq,ram 4/64gb,snap 845,lcd...    False      False  

[1802 rows x 4 columns]

In [484]:
compare_name_copy_2

name_detail  \
0     điện thoại may mắn, lung linh phù hợp các dòng...   
1     điện thoại vivo y20 (2021) 4gb + 64gb - hàng c...   
2                                  apple iphone 11 64gb   
3     điện thoại nokia1280 zin chính hãng đủ phụ kiệ...   
4     điện thoại samsung galaxy m12 (3gb/32gb) - hãn...   
...                                                 ...   
1797  điện thoại samsung galaxy note 20 ultra [chính...   
1798  yêu thích\n[mã 1010elsale1 giảm 5% đơn 3tr] lg...   
1799                                     mic nokia 6300   
1800  điện thoại xiaomi mi 11 lite 5g ram 8g/128g - ...   
1801                               điện thoại - beetech   

                                           name_overall  compare  
0     điện thoại may mắn, lung linh phù hợp các dòng...    False  
1     điện thoại vivo y20 (2021) 4gb + 64gb - hàng c...    False  
2                                  apple iphone 11 64gb     True  
3     điện thoại nokia1280 zin chính hãng đủ phụ kiệ...     True  
4     điện thoại samsung galaxy m12 (3gb/32gb) - hãn...     True  
...                                                 ...      ...  
1797  điện thoại samsung galaxy s20 fe 5g nguyên zin...    False  
1798   điện thoại mini 4g palm phone pvg100 new sealbox    False  
1799                         điện thoại oppo a3s vỡ màn    False  
1800                        điện thoại xiaomi mi 8 lite    False  
1801  điện thoại lg g7 thinq,ram 4/64gb,snap 845,lcd...    False  

[1802 rows x 3 columns]

In [493]:
compare_name_copy_2['name_detail'] = compare_name_copy_2['name_detail'].apply(clean_name)
compare_name_copy_2['name_overall'] = compare_name_copy_2['name_overall'].apply(clean_name)

In [494]:
compare_name_copy_2['compare_2'] = compare_name_copy_2['name_detail'] == compare_name_copy_2['name_overall']

In [495]:
compare_name_copy_2['compare_2'].value_counts()

True     985
False    817
Name: compare_2, dtype: int64

Apply lên data set

Detail data

In [514]:
detailed_data_copy = detailed_data.copy()

In [515]:
detailed_data_copy.drop(columns=['Unnamed: 0'], inplace=True)

In [518]:
detailed_data_copy.rename(columns={'name_product': 'name'}, inplace=True)

In [519]:
detailed_data_copy['name'] = detailed_data_copy['name'].apply(clean_name)
detailed_data_copy

name  rating_product  \
0    điện thoại may mắn, lung linh phù hợp các dòng...            4.83   
1    điện thoại vivo y20 (2021) 4gb + 64gb - hàng c...            4.95   
2                                 apple iphone 11 64gb            4.84   
3    điện thoại nokia1280 zin chính hãng đủ phụ kiệ...            5.00   
4    điện thoại samsung galaxy m12 (3gb/32gb) - hãn...            4.93   
..                                                 ...             ...   
107  điện thoại samsung galaxy note 20 ultra [chính...             NaN   
108  lg v50 thinq 5g (6gb|128gb) ,hàn quốc,snap 855...            4.33   
109                                     mic nokia 6300            5.00   
110  điện thoại xiaomi mi 11 lite 5g ram 8g/128g - ...             NaN   
111                               điện thoại - beetech            2.00   

     rating_count_product quantity_5_star  quantity_4_star  quantity_3_star  \
0                   632.0              81             11.0              NaN   
1                   940.0             916              7.0              NaN   
2                  1144.0            1,1k             32.0              NaN   
3                     4.0               4              0.0              NaN   
4                   732.0             703              6.0              NaN   
..                    ...             ...              ...              ...   
107                   NaN             NaN              NaN              NaN   
108                   3.0             2.0              0.0              NaN   
109                   2.0             NaN              NaN              NaN   
110                   NaN             NaN              NaN              NaN   
111                   1.0             0.0              0.0              NaN   

     quantity_2_star  quantity_1_star count_cmt  images_video  ...  \
0                NaN              NaN       360         293.0  ...   
1                NaN              NaN       394         292.0  ...   
2                NaN              NaN       614         511.0  ...   
3                NaN              NaN         3           2.0  ...   
4                NaN              NaN       334         276.0  ...   
..               ...              ...       ...           ...  ...   
107              NaN              NaN       NaN           NaN  ...   
108              NaN              NaN       2.0           1.0  ...   
109              NaN              NaN       NaN           NaN  ...   
110              NaN              NaN       NaN           NaN  ...   
111              NaN              NaN       0.0           0.0  ...   

               shopee_guaranteed                   company  \
0    3 Ngày Trả Hàng / Hoàn Tiền       Tổng kho hàng China   
1                            NaN       VIVO OFFICIAL STORE   
2                            NaN      Apple Flagship Store   
3    3 Ngày Trả Hàng / Hoàn Tiền          dienthoaicodocla   
4                            NaN     SAMSUNG OFFICIAL SHOP   
..                           ...                       ...   
107  3 Ngày Trả Hàng / Hoàn Tiền            VK Mobile Shop   
108  3 Ngày Trả Hàng / Hoàn Tiền         CHAT_Mobile Store   
109  3 Ngày Trả Hàng / Hoàn Tiền  LINH KIỆN ĐIỆN THOẠI PRO   
110  3 Ngày Trả Hàng / Hoàn Tiền            VK Mobile Shop   
111  3 Ngày Trả Hàng / Hoàn Tiền              Miumiu stose   

                  shop_name  shopee_mall  shop_total_review total_products  \
0           hung_pham130217        False               1,5k             13   
1    vivo_official_store_vn         True               6,6k             28   
2      apple_flagship_store         True              18,3k            128   
3      nokiae72zinchinhhang        False                 23             29   
4    samsung_official_store         True              39,3k            367   
..                      ...          ...                ...            ...   
107            phambaoloc09        False                302            701   
10

Overall data

In [500]:
overall_data = df_overall.copy()

In [503]:
overall_data.head()

name   price_dis  \
0  điện thoại may mắn, lung linh phù hợp các dòng...       1.400   
1  điện thoại vivo y20 (2021) 4gb + 64gb - hàng c...   3.390.000   
2                               apple iphone 11 64gb  13.990.000   
3  điện thoại nokia1280 zin chính hãng đủ phụ kiệ...     239.000   
4  điện thoại samsung galaxy m12 (3gb/32gb) - hãn...   3.490.000   

      price_nondis  freeship  rating  rating_count monthly_sold  \
0  ₫1.400 - ₫6.700      True    4.83         616.0   5,4k/tháng   
1       ₫3.990.000      True    4.95         926.0   1,8k/tháng   
2      ₫21.990.000      True    4.84        1132.0   1,1k/tháng   
3         ₫239.000     False    5.00           4.0    763/tháng   
4       ₫3.490.000      True    4.93         715.0    715/tháng   

    store_location  monthly_event favorite_product  shopee_mall discount_tag  \
0          Lào Cai          False        Yêu thích        False          NaN   
1  TP. Hồ Chí Minh           True              NaN         True    15%\nGIẢM   
2  TP. Hồ Chí Minh           True              NaN         True    36%\nGIẢM   
3         Hưng Yên          False              NaN        False          NaN   
4  TP. Hồ Chí Minh           True              NaN         True          NaN   

                                         product_url  \
0  https://shopee.vn/Mi%E1%BA%BFng-d%C3%A1n-%C4%9...   
1  https://shopee.vn/%C4%90i%E1%BB%87n-tho%E1%BA%...   
2  https://shopee.vn/Apple-iPhone-11-64GB-i.88201...   
3  https://shopee.vn/-H%C3%A0ng-Ch%C3%ADnh-H%C3%A...   
4  https://shopee.vn/%C4%90i%E1%BB%87n-Tho%E1%BA%...   

                                               image  ShopDacBiet  
0  https://cf.shopee.vn/file/cd7f36d677941ad531a8...        False  
1  https://cf.shopee.vn/file/41694acce229f8102e75...        False  
2  https://cf.shopee.vn/file/b107f2e6167adb250100...        False  
3  https://cf.shopee.vn/file/5d35156b434e236ee552...        False  
4  https://cf.shopee.vn/file/a277200d45df2ef46c4a...        False

In [502]:
overall_data['name'] = overall_data['name'].apply(clean_name)

Merge 2 dataframe 

Set index before concat

In [520]:
detailed_data_copy.set_index(['name'], inplace=True)

In [541]:
detailed_data_copy.head()

rating_product  \
name                                                                 
điện thoại may mắn, lung linh phù hợp các dòng máy            4.83   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...            4.95   
apple iphone 11 64gb                                          4.84   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...            5.00   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...            4.93   

                                                    rating_count_product  \
name                                                                       
điện thoại may mắn, lung linh phù hợp các dòng máy                 632.0   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...                 940.0   
apple iphone 11 64gb                                              1144.0   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...                   4.0   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...                 732.0   

                                                   quantity_5_star  \
name                                                                 
điện thoại may mắn, lung linh phù hợp các dòng máy              81   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...             916   
apple iphone 11 64gb                                          1,1k   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...               4   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...             703   

                                                    quantity_4_star  \
name                                                                  
điện thoại may mắn, lung linh phù hợp các dòng máy             11.0   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...              7.0   
apple iphone 11 64gb                                           32.0   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...              0.0   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...              6.0   

                                                    quantity_3_star  \
name                                                                  
điện thoại may mắn, lung linh phù hợp các dòng máy              NaN   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...              NaN   
apple iphone 11 64gb                                            NaN   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...              NaN   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...              NaN   

                                                    quantity_2_star  \
name                                                                  
điện thoại may mắn, lung linh phù hợp các dòng máy              NaN   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...              NaN   
apple iphone 11 64gb                                            NaN   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...              NaN   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...              NaN   

                                                    quantity_1_star count_cmt  \
name                                                                            
điện thoại may mắn, lung linh phù hợp các dòng máy              NaN       360   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...              NaN       394   
apple iphone 11 64gb                                            NaN       614   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...              NaN         3   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...              NaN       334   

                                                    images_video sold_out  \
name                                                                        
điện thoại may mắn, lung linh phù hợp các dòng máy         293.0    15,5k   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...         292.0     4,7k   
apple iphone 11 64gb                                       511.0     3,4k   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...           2.0      765   
điện thoại sa

In [507]:
overall_data.set_index(['name'], inplace=True)

In [540]:
overall_data.head()

price_dis  \
name                                                             
điện thoại may mắn, lung linh phù hợp các dòng máy       1.400   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...   3.390.000   
apple iphone 11 64gb                                13.990.000   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...     239.000   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...   3.490.000   

                                                       price_nondis  freeship  \
name                                                                            
điện thoại may mắn, lung linh phù hợp các dòng máy  ₫1.400 - ₫6.700      True   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...       ₫3.990.000      True   
apple iphone 11 64gb                                    ₫21.990.000      True   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...         ₫239.000     False   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...       ₫3.490.000      True   

                                                    rating  rating_count  \
name                                                                       
điện thoại may mắn, lung linh phù hợp các dòng máy    4.83         616.0   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...    4.95         926.0   
apple iphone 11 64gb                                  4.84        1132.0   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...    5.00           4.0   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...    4.93         715.0   

                                                   monthly_sold  \
name                                                              
điện thoại may mắn, lung linh phù hợp các dòng máy   5,4k/tháng   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...   1,8k/tháng   
apple iphone 11 64gb                                 1,1k/tháng   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...    763/tháng   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...    715/tháng   

                                                     store_location  \
name                                                                  
điện thoại may mắn, lung linh phù hợp các dòng máy          Lào Cai   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...  TP. Hồ Chí Minh   
apple iphone 11 64gb                                TP. Hồ Chí Minh   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...         Hưng Yên   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...  TP. Hồ Chí Minh   

                                                    monthly_event  \
name                                                                
điện thoại may mắn, lung linh phù hợp các dòng máy          False   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...           True   
apple iphone 11 64gb                                         True   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...          False   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...           True   

                                                   favorite_product  \
name                                                                  
điện thoại may mắn, lung linh phù hợp các dòng máy        Yêu thích   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...              NaN   
apple iphone 11 64gb                                            NaN   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...              NaN   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...              NaN   

                                                    shopee_mall discount_tag  \
name                                                                           
điện thoại may mắn, lung linh phù hợp các dòng máy        False          NaN   
điện thoại vivo y20 (2021) 4gb + 64gb - hàng ch...         True    15%\nGIẢM   
apple iphone 11 64gb                                       True    36%\nGIẢM   
điện thoại nokia1280 zin chính hãng đủ phụ kiện...        False          NaN   
điện thoại samsung galaxy m12 (3gb/32gb) - hãng...         

In [539]:
pd.merge(overall_data, detailed_data_copy, how='inner', left_index= True, right_index=True)

price_dis price_nondis  \
name                                                                        
(có pin sạc) nokia 101- 2 sim nghe nhạc bảo hàn...   135.000     ₫200.000   
(có pin, sạc)nokia 1280 zin chính hãng lên sườn...   175.000     ₫180.000   
1 caphe idol gc loại mạnh, mẫu cũ.1h 10goi.          120.000     ₫120.000   
[ chính hãng ] tai nghe oppo f3,f5,f7,f9,f11,r1...    50.000      ₫90.000   
[ giá siêu rẻ,chính hãng 100% ] bàn phím thay t...    15.000      ₫15.000   
...                                                      ...          ...   
✅ chính hãng ✅ kính camera sau xiaomi mi 10t pr...    69.000      ₫69.000   
✅ chính hãng ✅ loa ngoài xiaomi redmi 4 prime c...    99.000      ₫99.000   
✅ chính hãng ✅ nắp lưng vỏ kèm sạc samsung gear...   149.000     ₫149.000   
✅ chính hãng ✅ nắp lưng vỏ sau xiaomi redmi k30...   180.000     ₫180.000   
🌟 loa ngoài chuông 🌟 oppo realme c1 chính hãng       150.000     ₫150.000   

                                                    freeship_x  rating  \
name                                                                     
(có pin sạc) nokia 101- 2 sim nghe nhạc bảo hàn...        True    4.68   
(có pin, sạc)nokia 1280 zin chính hãng lên sườn...       False    5.00   
1 caphe idol gc loại mạnh, mẫu cũ.1h 10goi.               True    5.00   
[ chính hãng ] tai nghe oppo f3,f5,f7,f9,f11,r1...        True    4.92   
[ giá siêu rẻ,chính hãng 100% ] bàn phím thay t...        True    4.50   
...                                                        ...     ...   
✅ chính hãng ✅ kính camera sau xiaomi mi 10t pr...       False    5.00   
✅ chính hãng ✅ loa ngoài xiaomi redmi 4 prime c...       False     NaN   
✅ chính hãng ✅ nắp lưng vỏ kèm sạc samsung gear...       False    5.00   
✅ chính hãng ✅ nắp lưng vỏ sau xiaomi redmi k30...       False     NaN   
🌟 loa ngoài chuông 🌟 oppo realme c1 chính hãng            True     NaN   

                                                    rating_count monthly_sold  \
name                                                                            
(có pin sạc) nokia 101- 2 sim nghe nhạc bảo hàn...         119.0     34/tháng   
(có pin, sạc)nokia 1280 zin chính hãng lên sườn...           1.0      3/tháng   
1 caphe idol gc loại mạnh, mẫu cũ.1h 10goi.                  3.0      2/tháng   
[ chính hãng ] tai nghe oppo f3,f5,f7,f9,f11,r1...         545.0    199/tháng   
[ giá siêu rẻ,chính hãng 100% ] bàn phím thay t...           2.0      1/tháng   
...                                                          ...          ...   
✅ chính hãng ✅ kính camera sau xiaomi mi 10t pr...           2.0      4/tháng   
✅ chính hãng ✅ loa ngoài xiaomi redmi 4 prime c...           NaN      1/tháng   
✅ chính hãng ✅ nắp lưng vỏ kèm sạc samsung gear...           1.0      1/tháng   
✅ chính hãng ✅ nắp lưng vỏ sau xiaomi redmi k30...           NaN      1/tháng   
🌟 loa ngoài chuông 🌟 oppo realme c1 chính hãng               NaN      1/tháng   

                                                     store_location  \
name                                                                  
(có pin sạc) nokia 101- 2 sim nghe nhạc bảo hàn...           Hà Nội   
(có pin, sạc)nokia 1280 zin chính hãng lên sườn...           Hà Nội   
1 caphe idol gc loại mạnh, mẫu cũ.1h 10goi.         TP. Hồ Chí Minh   
[ chính hãng ] tai nghe oppo f3,f5,f7,f9,f11,r1...           Hà Nội   
[ giá siêu rẻ,chính hãng 100% ] bàn phím thay t...           Hà Nội   
...                                                             ...   
✅ chính hãng ✅ kính camera sau xiaomi mi 10t pr...  TP. Hồ Chí Minh   
✅ chính hãng ✅ loa ngoài xiaomi redmi 4 prime c...  TP. Hồ Chí Minh   
✅ chính hãng ✅ nắp lưng vỏ kèm sạc samsung gear...  TP. Hồ Chí Minh   
✅ chính hãng ✅ nắp lưng vỏ sau xiaomi redmi k30...  TP. Hồ Chí Minh   
🌟 loa ngoài chuông 🌟 oppo realme c1 chính hãng      TP. Hồ Chí Minh   

                                                    monthly_event  \
name                                   

In [543]:
test_merge = pd.read_csv('/Users/Thome/Desktop/CoderSchool/detailed_data_test.csv')

In [546]:
pd.merge(overall_data, test_merge, how='inner', left_on='product_url', right_on='product_url').T

0
price_dis                                                         1.400
price_nondis                                            ₫1.400 - ₫6.700
freeship_x                                                         True
rating                                                             4.83
rating_count                                                      616.0
monthly_sold                                                 5,4k/tháng
store_location                                                  Lào Cai
monthly_event                                                     False
favorite_product                                              Yêu thích
shopee_mall_x                                                     False
discount_tag                                                        NaN
product_url           https://shopee.vn/Mi%E1%BA%BFng-d%C3%A1n-%C4%9...
image                 https://cf.shopee.vn/file/cd7f36d677941ad531a8...
ShopDacBiet                                                       False
Unnamed: 0                                                            0
name_product          Yêu Thích\nMiếng dán điện thoại may mắn, lung ...
rating_product                                                     4.83
rating_count_product                                                669
quantity_5_star                                                      61
quantity_4_star                                                      12
quantity_3_star                                                     NaN
quantity_2_star                                                     NaN
quantity_1_star                                                     NaN
count_cmt                                                           380
images_video                                                        311
sold_out                                                          16,4k
low_price                                                        6700.0
hight_price                                                      1400.0
price                                                               NaN
inventory                                                         21739
return_7_days                                                       NaN
authentic                                                           NaN
freeship_y                                                          NaN
shopee_guaranteed                           3 Ngày Trả Hàng / Hoàn Tiền
company                                             Tổng kho hàng China
shop_name                                               hung_pham130217
shopee_mall_y                                                     False
shop_total_review                                                  1,5k
total_products                                                       14
response_rate                                                       97%
time_response                                             trong vài giờ
years_ex                                                 28 tháng trước
follower                                                           3,3k

In [547]:
pd.read_csv('/Users/Thome/Desktop/CoderSchool/detailed_data_500.csv')

Unnamed: 0                                        product_url  \
0             0  https://shopee.vn/Mi%E1%BA%BFng-d%C3%A1n-%C4%9...   
1             1  https://shopee.vn/%C4%90i%E1%BB%87n-tho%E1%BA%...   
2             2  https://shopee.vn/Apple-iPhone-11-64GB-i.88201...   
3             3  https://shopee.vn/-H%C3%A0ng-Ch%C3%ADnh-H%C3%A...   
4             4  https://shopee.vn/%C4%90i%E1%BB%87n-Tho%E1%BA%...   
..          ...                                                ...   
436         436  https://shopee.vn/%C4%90i%E1%BB%87n-tho%E1%BA%...   
437         437  https://shopee.vn/%C4%90i%E1%BB%87n-tho%E1%BA%...   
438         438  https://shopee.vn/%C4%90i%C3%AA%CC%A3n-thoa%CC...   
439         439  https://shopee.vn/Clear-kho-%C4%90i%E1%BB%87n-...   
440         440  https://shopee.vn/%C4%90i%E1%BB%87n-tho%E1%BA%...   

                                          name_product  rating_product  \
0    Yêu Thích\nMiếng dán điện thoại may mắn, lung ...            4.83   
1    Điện thoại Vivo Y20 (2021) 4GB + 64GB - Hàng c...            4.95   
2                                 Apple iPhone 11 64GB            4.83   
3    [Hàng Chính Hãng 100%] Nokia 1280, Điện Thoại ...            4.83   
4    Điện Thoại Samsung Galaxy M12 (3GB/32GB) - Hãn...            4.92   
..                                                 ...             ...   
436  Điện thoại Mastel IZI125 2 sim 2 sóng online p...            5.00   
437  Yêu Thích\nĐiện thoại 3 sim 3 sóng online Q4 -...            5.00   
438  Điện thoại poco x3 pro 6.67" snapdragon 860 ...            5.00   
439  Yêu Thích+\nClear kho Điện thoại Philips E106 ...            4.08   
440  Yêu Thích\nĐiện thoại Sharp Aquos R2 Compact -...            5.00   

     rating_count_product quantity_5_star  quantity_4_star  quantity_3_star  \
0                   673.0             619             12.0              NaN   
1                   978.0             954              7.0              NaN   
2                  1184.0             NaN              NaN              NaN   
3                     6.0             NaN              NaN              NaN   
4                   763.0             732              6.0              NaN   
..                    ...             ...              ...              ...   
436                   2.0               2              0.0              NaN   
437                   5.0             NaN              0.0              NaN   
438                  18.0              18              0.0              NaN   
439                  13.0              10              3.0              NaN   
440                   4.0               4              0.0              NaN   

     quantity_2_star  quantity_1_star  ...            shopee_guaranteed  \
0                NaN              NaN  ...  3 Ngày Trả Hàng / Hoàn Tiền   
1                NaN              NaN  ...                          NaN   
2                NaN              NaN  ...                          NaN   
3                NaN              NaN  ...  3 Ngày Trả Hàng / Hoàn Tiền   
4                NaN              NaN  ...                          NaN   
..               ...              ...  ...                          ...   
436              NaN              NaN  ...  3 Ngày Trả Hàng / Hoàn Tiền   
437              NaN              NaN  ...  3 Ngày Trả Hàng / Hoàn Tiền   
438              NaN              NaN  ...  3 Ngày Trả Hàng / Hoàn Tiền   
439              NaN              NaN  ...  3 Ngày Trả Hàng / Hoàn Tiền   
440              NaN              NaN  ...  3 Ngày Trả Hàng / Hoàn Tiền   

                     company               shop_name  shopee_mall  \
0        Tổng kho hàng China         hung_pham130217        False   
1        VIVO OFFICIAL STORE  vivo_official_store_vn         True   
2       Apple Flagship Store    apple_flagship_store         True   
3           dienthoaicodocla    nokiae72zinchinhhang        False   
4      SAMSUNG OFFICIAL SHOP  samsung_official_store         True   
..         